src/udev/net/ethtool-util.c

```diff
@@ -323,3 +265,268 @@ int ethtool_set_features(int *fd, const char *ifname, NetDevFeature *features) {
 
         return 0;
 }
+
+#ifndef HAVE_STRUCT_ETHTOOL_LINK_SETTINGS
```

Comment:
This should not be compile time dependent. The code should dynamically fall back to this function if the new code fails. After all, it should be perfectly legal to compile on a newer kernel and run on an older one.